packages/components/src/dashicon/index.js

```diff
@@ -20,7 +22,17 @@ function Dashicon( { icon, className, ...extraProps } ) {
 		.filter( Boolean )
 		.join( ' ' );
 
-	return <span className={ iconClass } { ...extraProps } />;
+	// For retro-compatibility reasons (for example if people are overriding icon size with CSS), we add inline styles just if the size is different to the default
+	const style =
+		20 !== size
+			? {
+					fontSize: `${ size }px`,
+					width: `${ size }px`,
+					height: `${ size }px`,
+			  }
+			: {};
+
+	return <span className={ iconClass } style={ style } { ...extraProps } />;
```

Comment:
I think we need to merge the incoming `style` prop so it works as before.

```jsx
// So something like this still works
<Dashicon icon="foo" style={{ background: 'yellow' }} />
```

Reply:
Goog catch, thanks! Done.